gateway/crates/federated-server/src/server/gateway.rs

```diff
@@ -64,12 +65,30 @@ pub(super) async fn generate(
         runtime::trusted_documents_client::Client::new(NoopTrustedDocuments)
     };
 
+    let mut redis_factory = RedisPoolFactory::default();
+
     let watcher = ConfigWatcher::init(gateway_config.clone(), hot_reload_config_path)?;
 
-    let rate_limiter = match &gateway_config.gateway.rate_limit {
-        Some(config) if config.storage.is_redis() => RedisRateLimiter::runtime(&config.redis, watcher)
-            .await
-            .map_err(|e| crate::Error::InternalError(e.to_string()))?,
+    let rate_limiter = match config.rate_limit_config() {
+        Some(config) if config.storage.is_redis() => {
+            let tls = config.redis.tls.map(|tls| RedisTlsConfig {
+                cert: tls.cert,
+                key: tls.key,
+                ca: tls.ca,
+            });
+
+            let pool = redis_factory
+                .pool(config.redis.url, tls)
```

Comment:
:zap: **Suggestion** 
Ensure that the error handling for the Redis connection pool creation is robust and provides meaningful feedback.